### PR TITLE
feat : update platforms list and the --platform option

### LINKED ISSUE #234: The `macos` alias for `osx` is not supported
```
$ tldr --platform macos
usage: tldr command [options]
tldr: error: argument -p/--platform: invalid choice: 'macos' (choose from 'linux', 'osx', 'sunos', 'windows', 'common')
```
https://github.com/tldr-pages/tldr/blob/main/CLIENT-SPECIFICATION.md
> It is RECOMMENDED that clients support `macos` as an alias for `osx`.

### PR DESCRIPTION
**Pull request by patricedenis**
This PR may close #234, #235 and #236.
Be aware that this pull request is based only on code analysis and has not been tested!
I would be really pleased if someone could test this PR and see if it works.

### CODE REVIEW

**Comment by patricedenis** [comment]
I'm sorry I didn't know how to continue the work that was done on #236.

**Comment by patricedenis** [comment]
It's nice to see that the workflow is checking my code so I can fix it. Good feature !

File: tldr.py [hunk 1]
```diff
@@ -41,14 +41,18 @@
     URLOPEN_CONTEXT.check_hostname = False
     URLOPEN_CONTEXT.verify_mode = ssl.CERT_NONE
 
-OS_DIRECTORIES = {
-    "linux": "linux",
+ OS_DIRECTORIES = {
```
Comment (spageektti) [inline tldr.py:44]:
```suggestion
OS_DIRECTORIES = {
```
Is the space a typo?

Comment (patricedenis) [reply tldr.py:44]: let's test it

File: tldr.py [hunk 2]
```diff
@@ -42,13 +42,17 @@
     URLOPEN_CONTEXT.verify_mode = ssl.CERT_NONE
 
 OS_DIRECTORIES = {
-    "linux": "linux",
+    "android": "android",
     "darwin": "osx",
     "freebsd": "freebsd",
-    "openbsd": "openbsd",
+    "linux": "linux",
+    "macos": "osx"
```
Comment (spageektti) [inline tldr.py:49]:
```suggestion
    "macos": "osx",
```

**Comment by patricedenis** [comment]
I don't understand the lint codebase message :

```
Run python3 -m flake8
./tldr.py:522:89: E501 line too long (106 > 88 characters)
./tldr.py:524:89: E501 line too long (116 > 88 characters)
Error: Process completed with exit code 1.
```
I opened the raw file in a text editor and did not see the supposed added characters on lines 522 and 524.

**Comment by patricedenis** [comment]
Now there's a e131 lint error.
it's again related to indentation.

**Review by spageektti — APPROVED** [review]
LGTM but havent tested locally

**Review by kbdharun — APPROVED** [review]
LGTM, Thanks for your contribution. Tested the changes locally, this PR is GTG.

**Pull request merged**